src/features/environmentDetails/components/EnvironmentDetails.tsx

```diff
@@ -34,30 +38,67 @@ import artifactList from "../../../utils/helpers/artifact";
 import createLabel from "../../../common/config/labels";
 import { AlertDialog } from "../../../components/Dialog";
 import { useAppDispatch, useAppSelector } from "../../../hooks";
-import { CondaSpecificationPip } from "../../../common/models";
+import {
+  CondaSpecificationPip,
+  Environment,
+  Namespace
+} from "../../../common/models";
 import { useInterval } from "../../../utils/helpers";
+import { showNotification } from "../../notification/notificationSlice";
+import { useScrollRef } from "../../../layouts/PageLayout";
 
-interface IEnvDetails {
-  scrollRef: any;
-  environmentNotification: (notification: any) => void;
-}
 interface IUpdateEnvironmentArgs {
   dependencies: (string | CondaSpecificationPip)[];
   channels: string[];
 }
 
 const INTERVAL_REFRESHING = 5000;
 
-export const EnvironmentDetails = ({
-  scrollRef,
-  environmentNotification
-}: IEnvDetails) => {
+export const EnvironmentDetails = () => {
   const dispatch = useAppDispatch();
+
+  // Url routing params
+  // If user loads the app at /<namespace_name>/<environment_name>
+  // This will put the app in the correct state
+  const { namespaceName, environmentName } = useParams<{
+    namespaceName: string;
+    environmentName: string;
+  }>();
+  const namespaces: Namespace[] = useAppSelector(
+    state => state.namespaces.data
+  );
+  const namespace = namespaces.find(({ name }) => name === namespaceName);
+  const foundNamespace = Boolean(namespace);
+  const environments: Environment[] = useAppSelector(
+    state => state.environments.data
+  );
+  const environment = environments.find(
+    environment =>
+      environment.namespace.name === namespaceName &&
+      environment.name === environmentName
+  );
+  const foundEnvironment = Boolean(environment);
+  useEffect(() => {
+    if (namespace && environment) {
+      dispatch(
+        environmentOpened({
+          environment,
+          canUpdate: namespace.canUpdate
+        })
+      );
+      dispatch(modeChanged(EnvironmentDetailsModes.READ));
+      dispatch(toggleNewEnvironmentView(false));
+    }
+  }, [namespaceName, environmentName, foundNamespace, foundEnvironment]);
+
+  const navigate = useNavigate();
+
   const { mode } = useAppSelector(state => state.environmentDetails);
   const { page, dependencies } = useAppSelector(state => state.dependencies);
   const { selectedEnvironment } = useAppSelector(state => state.tabs);
   const { currentBuild } = useAppSelector(state => state.enviroments);
   const [name, setName] = useState(selectedEnvironment?.name || "");
+  const scrollRef = useScrollRef();
```

Comment:
The scroll ref was being passed in by the parent previously, but now the parent cannot pass down properties directly to the child components because it's handled by React Router. So we need to use the `useOutletContext` hook, and we wrap it in another hook (useScrollRef) for better type checking.